pandas/core/generic.py

```diff
@@ -6306,10 +6306,21 @@ def fillna(
                 return result if not inplace else None
 
             elif not is_list_like(value):
-                new_data = self._mgr.fillna(
-                    value=value, limit=limit, inplace=inplace, downcast=downcast
-                )
+                if not self._mgr.is_single_block and axis == 1:
```

Comment:
isn't this the same as L6254

Reply:
Yes, the code is somehow the same, with minor adjustments. I basically used it to make transposes as the ones done when the method is valid. This is because when the method is valid, both `axis = 1` and `axis = 0` works correctly with the given limit for `Fillna` due to the transposition at L6254. Therefore, I tried to do the same kind of transposing so that `axis = 1` and `axis = 0` would work accordingly for the given value and limit.

Reply:
ok any possibiltiy of combing those cases into a similar code block?